From e1f85fe0cdfa7e4d4ce7a811d0b0c90bf38fba0c Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 25 Jun 2009 09:47:27 +1000
Subject: [PATCH] Blacklist HDAPS accelerometer device from being picked up in X.

This device posts accelerometer data through ABS_X/ABS_Y, making X unusable
if it's controlling the pointer.
---
 fdi/policy/10osvendor/10-x11-input.fdi |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/fdi/policy/10osvendor/10-x11-input.fdi b/fdi/policy/10osvendor/10-x11-input.fdi
index a342421..89ba672 100644
--- a/fdi/policy/10osvendor/10-x11-input.fdi
+++ b/fdi/policy/10osvendor/10-x11-input.fdi
@@ -38,5 +38,11 @@
         <merge key="input.x11_driver" type="string">evdev</merge>
       </match>
     </match>
+
+    <!-- http://bugs.freedesktop.org/show_bug.cgi?id=22442 
+         Posts accel data through ABS_X/ABS_Y, makes X unusable -->
+    <match key="info.product" contains="ThinkPad HDAPS accelerometer data">
+        <remove key="input.x11_driver" />
+    </match>
   </device>
 </deviceinfo>
-- 
1.6.3.rc1.2.g0164.dirty

